ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFArrayIntersect.java

```diff
@@ -43,19 +47,27 @@ public GenericUDFArrayIntersect() {
     ObjectInspector defaultOI = super.initialize(arguments);
     checkArgCategory(arguments, ARRAY2_IDX, ObjectInspector.Category.LIST, FUNC_NAME,
         org.apache.hadoop.hive.serde.serdeConstants.LIST_TYPE_NAME); //Array1 is already getting validated in Parent class
+    if (!ObjectInspectorUtils.compareTypes(arrayOI.getListElementObjectInspector(), ((ListObjectInspector) arguments[ARRAY2_IDX]).getListElementObjectInspector())) { // check if elements of arrays are comparable
+      throw new UDFArgumentTypeException(1, ERROR_NOT_COMPARABLE);
+    }
     return defaultOI;
   }
 
   @Override public Object evaluate(DeferredObject[] arguments) throws HiveException {
 
     Object array = arguments[ARRAY_IDX].get();
+    Object array2 = arguments[ARRAY2_IDX].get();
+    if (array == null) {
```

Comment:
I would change it for if (array == null || array2 == null)  {  }  I don't see the value to have separate IF statement

Reply:
Changed as recommended